tools/pythonpkg/src/python_conversion.cpp

```diff
@@ -201,59 +201,122 @@ bool TryTransformPythonNumeric(Value &res, py::handle ele) {
 	return true;
 }
 
-Value TransformPythonValue(py::handle ele, const LogicalType &target_type, bool nan_as_null) {
+PythonObjectType GetObjectType(py::handle &ele) {
 	auto &import_cache = *DuckDBPyConnection::ImportCache();
 
 	if (ele.is_none()) {
-		return Value();
+		return PythonObjectType::None;
 	} else if (py::isinstance<py::bool_>(ele)) {
-		return Value::BOOLEAN(ele.cast<bool>());
+		return PythonObjectType::Bool;
 	} else if (py::isinstance<py::int_>(ele)) {
+		return PythonObjectType::Integer;
+	} else if (py::isinstance<py::float_>(ele)) {
+		return PythonObjectType::Float;
+	} else if (py::isinstance(ele, import_cache.decimal.Decimal())) {
+		return PythonObjectType::Decimal;
+	} else if (py::isinstance(ele, import_cache.uuid.UUID())) {
+		return PythonObjectType::Uuid;
+	} else if (py::isinstance(ele, import_cache.datetime.datetime())) {
+		return PythonObjectType::Datetime;
+	} else if (py::isinstance(ele, import_cache.datetime.time())) {
+		return PythonObjectType::Time;
+	} else if (py::isinstance(ele, import_cache.datetime.date())) {
+		return PythonObjectType::Date;
+	} else if (py::isinstance(ele, import_cache.datetime.timedelta())) {
+		return PythonObjectType::Timedelta;
+	} else if (py::isinstance<py::str>(ele)) {
+		return PythonObjectType::String;
+	} else if (py::isinstance<py::bytearray>(ele)) {
+		return PythonObjectType::ByteArray;
+	} else if (py::isinstance<py::memoryview>(ele)) {
+		return PythonObjectType::MemoryView;
+	} else if (py::isinstance<py::bytes>(ele)) {
+		return PythonObjectType::Bytes;
+	} else if (py::isinstance<py::list>(ele)) {
+		return PythonObjectType::List;
+	} else if (py::isinstance<py::dict>(ele)) {
+		return PythonObjectType::Dict;
+	} else if (py::isinstance(ele, import_cache.numpy.ndarray())) {
+		return PythonObjectType::NdArray;
+	} else {
+		return PythonObjectType::Other;
+	}
+}
+
+Value TransformPythonValue(py::handle ele, const LogicalType &target_type, bool nan_as_null) {
+	auto &import_cache = *DuckDBPyConnection::ImportCache();
+
+	auto object_type = GetObjectType(ele);
+	if (object_type != PythonObjectType::Date) {
+		;
```

Comment:
Is this supposed to be an empty statement?

Reply:
Ah crap, that's a leftover from debugging :sweat_smile: